lmdeploy/pytorch/kernels/pagedattention.py

```diff
@@ -446,22 +604,30 @@ def _kernel_meta():
 
     # shape constraints
     Lq, Lk, Lv = q.shape[-1], k.shape[-1], v.shape[-1]
-    assert Lq == Lk and Lk == Lv
+    assert Lq == Lk, Lv == o.shape[-1]
 
-    sm_scale = 1.0 / (Lq**0.5)
+    if sm_scale is None:
+        sm_scale = 1.0 / (Lq**0.5)
```

Comment:
Just out of curiosity, which model uses a different `sm_scale` other than `1/sqrt(dim)`

Reply:
https://huggingface.co/deepseek-ai/DeepSeek-V2-Lite-Chat/blob/main/modeling_deepseek.py#L749
And MLA might use the sm_scale of origin dim.